src/sources/kubernetes_logs/mod.rs

```diff
@@ -271,30 +272,31 @@ impl Source {
             futures::channel::mpsc::channel::<Vec<(Bytes, String)>>(2);
 
         let mut parser = parser::build();
-        let mut partial_events_merger = partial_events_merger::build(auto_partial_merge);
-
-        let events =
-            file_source_rx
-                .map(futures::stream::iter)
-                .flatten()
-                .map(move |(bytes, file)| {
-                    emit!(KubernetesLogsEventReceived {
-                        file: &file,
-                        byte_size: bytes.len(),
-                    });
-                    let mut event = create_event(bytes, &file);
-                    if annotator.annotate(&mut event, &file).is_none() {
-                        emit!(KubernetesLogsEventAnnotationFailed { event: &event });
-                    }
-                    event
-                });
+        let partial_events_merger = Box::new(partial_events_merger::build(auto_partial_merge));
+
+        let events = file_source_rx.map(futures::stream::iter);
+        let events = events.flatten();
+        let events = events.map(move |(bytes, file)| {
+            emit!(KubernetesLogsEventReceived {
+                file: &file,
+                byte_size: bytes.len(),
+            });
+            let mut event = create_event(bytes, &file);
+            if annotator.annotate(&mut event, &file).is_none() {
+                emit!(KubernetesLogsEventAnnotationFailed { event: &event });
+            }
+            event
+        });
         let events = events
-            .filter_map(move |event| futures::future::ready(parser.transform(event)))
-            .filter_map(move |event| {
-                futures::future::ready(partial_events_merger.transform(event))
+            .flat_map(move |event| {
+                let mut buf = Vec::with_capacity(1);
+                parser.transform(&mut buf, event);
+                futures::stream::iter(buf)
             });
 
-        let event_processing_loop = events.map(Ok).forward(out);
+        let event_processing_loop = partial_events_merger.transform(
+            Box::new(events.map(Ok).compat())
+        ).map_err(|_| unimplemented!("TODO")).compat().forward(out);
```

Comment:
Hey @MOZGIII , I'm wondering if you can help me with this one?

Now that the `partial_events_merger` is a `TaskTransform` it needs to be attached to the stream, but the `Sink::Error` seems to be restricted to `std::error::Error` and returning `()`? I fiddled around for a long time but I'm hoping you might have some context that could help me?

Reply:
Sure, I'll take a look!

Reply:
```suggestion
        ).map_err(|_| unreachable!("TODO: remove with futures 0.3 transition")).compat().forward(out);
```